portal-ui/src/screens/Console/Tools/KeyRevealer.tsx

```diff
@@ -0,0 +1,60 @@
+import { Box, Button } from "@mui/material";
+import InputBoxWrapper from "../Common/FormComponents/InputBoxWrapper/InputBoxWrapper";
+import { CopyIcon } from "../../../icons";
+import {connect} from "react-redux";
+import {AppState} from "../../../store";
+import {setErrorSnackMessage} from "../../../actions";
+import {useState} from "react";
+const KeyRevealer = ({ value }: { value: string }) => {
+  const [shown, setShown] = useState<boolean>(false);
+
+  return (
+      <Box
+          sx={{
+            display: "flex",
+            alignItems: "center",
+            flexFlow: {
+              sm: "row",
+              xs: "column",
+            },
+          }}
+      >
+        <InputBoxWrapper
+            id="inspect-dec-key"
+            name="inspect-dec-key"
+            placeholder=""
+            label=""
+            type={shown ? "text" : "password"}
+            onChange={() => {}}
+            value={value}
+            overlayIcon={<CopyIcon />}
+            extraInputProps={{
+              readOnly: true,
+            }}
+            overlayAction={() => navigator.clipboard.writeText(value)}
+        />
+
+        <Button
+            sx={{
+              marginLeft: "10px",
+            }}
+            variant="contained"
+            onClick={() => setShown(!shown)}
+        >
+          Show/Hide
+        </Button>
+      </Box>
+  );
+};
+
+const mapState = (state: AppState) => ({
+  distributedSetup: state.system.distributedSetup,
+});
+
+const mapDispatchToProps = {
+  setErrorSnackMessage,
+};
+
+const connector = connect(mapState, mapDispatchToProps);
+
+export default connector(KeyRevealer);
```

Comment:
Please remove the `setErrorSnackMessage` , `mapState`, `connect` as it is not required in this component.
```suggestion
export default KeyRevealer;
```